File: examples/Python/MyProjects/PySideDoodle/step01/step01.py
"""
// ============================================================================
// step01.py: creating the basic Doodle PyQt application
//
// Tutorial - PySide6 Doodle Application
// Based on a similar tutorial for Borland ObjectWindows Library (OWL)
// @author: Manish Bhobe
// My experiments with the Qt Framework. Use at your own risk!!
// ============================================================================
"""
import sys
import platform
import PySide6
from PySide6.QtGui import *
from PySide6.QtGui import *
from mainWindow import *

# code to import Chocolaf theme files
# from chocolaf.palettes import ChocolafPalette
# from chocolaf.utils.chocolafapp import Chocolaf

__version__ = "1.0"


def main():
    app = QApplication(sys.argv)
    print(f"PySide Doodle - running with Python {platform.python_version()}, " +
          f"Qt {PySide6.QtCore.__version__}, PySide {PySide6.__version__} on {platform.system()}")

    mainWindow = MainWindow()
    mainWindow.show()

    return app.exec()


if __name__ == "__main__":
    main()

File: examples/C++/MyProjects/Doodle/step01/step01.cc
// ============================================================================
// step01.cc: Creating a basic application with Qt/C++
//
// Tutorial - Qt Scribble Application
// Based on a similar tutorial for Borland ObjectWindows Library (OWL)
//
// @author Manish Bhobe for Nämostuté Ltd.
// My experiments with C++,Qt, Python & PyQt.
// Code is provided for illustration purposes only! Use at your own risk.
// =============================================================================
#include <QApplication>
#include <QMainWindow>
#include <QtGui>

#include "chocolaf.h"

int main(int argc, char **argv)
{
  QApplication app(argc, argv);
  Chocolaf::setChocolafStyle(app, "Chocolaf");

  // create the GUI
  QMainWindow mainWindow;
  QString title = QString("Qt %1 Doodle with Chocolaf - Step01: Basic Window").arg(QT_VERSION_STR);
  mainWindow.setWindowTitle(title);
  Chocolaf::centerOnScreenWithSize(mainWindow, 0.75, 0.75);
  mainWindow.show();

  return app.exec();
}
